flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -569,6 +571,69 @@ public int compare(JobStatusMessage o1, JobStatusMessage o2) {
 	}
 
 	/**
+	 * Executes the STOP action.
+	 * 
+	 * @param args Command line arguments for the stop action.
+	 */
+	protected int stop(String[] args) {
+		LOG.info("Running 'stop' command.");
+
+		StopOptions options;
+		try {
+			options = CliFrontendParser.parseStopCommand(args);
+		}
+		catch (CliArgsException e) {
+			return handleArgException(e);
+		}
+		catch (Throwable t) {
+			return handleError(t);
+		}
+
+		// evaluate help flag
+		if (options.isPrintHelp()) {
+			CliFrontendParser.printHelpForStop();
+			return 0;
+		}
+
+		String[] stopArgs = options.getArgs();
+		JobID jobId;
+
+		if (stopArgs.length > 0) {
+			String jobIdString = stopArgs[0];
+			try {
+				jobId = new JobID(StringUtils.hexStringToByte(jobIdString));
+			}
+			catch (Exception e) {
+				LOG.error("Error: The value for the Job ID is not a valid ID.");
```

Comment:
The thing is that `handleError` was introduced after the `cancel` method has been written. Thus at this time, it was not available, yet. So why should we keep the deficiencies of an old implementation if we could do it better? Of course, it would be nice to update `cancel` accordingly, but this is not part of this PR and should happen in a different PR.

IMHO, it is better to ditch antiquated traditions if you're implementing a new function.